src/main/webapp/js/stats-loader.js

```diff
@@ -1,23 +1,22 @@
-// Fetch stats and display them in the page.
-function fetchStats(){
-    const url = '/stats';
-    fetch(url).then((response) => {
+function fetchStats() {
+  const url = "/stats";
+  fetch(url)
+    .then(response => {
       return response.json();
-    }).then((stats) => {
-      const statsContainer = document.getElementById('stats-container');
-      statsContainer.innerHTML = '';
-      const messageCountElement = buildStatElement('Message count: ' + stats.messageCount);
+    })
+    .then(stats => {
+      const statsContainer = document.getElementById("stats-container");
```

Comment:
It's possible that the document will not have completed loading by this point, and thus the `stats-container` might not yet exist. I'd suggest the following minor refactor:


     var _statsPromise = null;
     function getOrPrefetchStats() {
       if (_statsPromise == null) {
         _statsPromise = fetch(url).then(response => { return response.json(); });
       }
       return _statsPromise;
     }

     function renderStatsOnPage() {
       getOrPrefetchStats().then(stats = > {
            const statsContainer = ...
            ...
       });
     }

And then in the HTML:

    ...
    <head>
       ...
     <script>getOrPrefetchStats();</script>
    </head>
    <body onload="renderStatsOnPage">
        ...
    </body>

Reply:
Perfect, this makes perfect sense to me now.